igc_lib.py

```diff
@@ -71,7 +73,148 @@ def _rawtime_float_to_hms(timef):
+                else:
+                    kind = "start_enter"
+            else:
+                if point == tpoints[-1]:   # if it is the last turnpoint i.e. the goal
+                    if point.getAttribute("type") == "line":
+                        kind = "goal_cylinder"     # to change once kind 'line' can be processed.
+                    else:
+                        kind = "goal_cylinder"
+                else:
+                    kind = "cylinder"
+            
+                      
+            turnpoint = Turnpoint(lat, lon, radius, kind)
+            turnpoints.append(turnpoint)
+        task = Task(turnpoints, start_time, end_time)
+        return task
+    
+    
+    def __init__(self, turnpoints, start_time, end_time):  #finish_time defaults to 23:59
+        self.turnpoints = turnpoints
+        self.start_time = start_time
+        self.end_time = end_time
+        
+        
+        
+    def check_flight(self, Flight):
+        """ Checks a Flight object against the task. 
+            Args:
+                   Flight: a Flight object
+            Returns:
+                    a list of GNSSFixes of when turnpoints were achieved.            
+        """
+        reached_turnpoints = []   
+        proceed_to_start = False
+        t=0
+        for fix in Flight.fixes:
+            if t >= len(self.turnpoints):
+                break  # pilot has arrived in goal (last turnpoint) so we can stop.
+        
+            #pilot must have at least 1 fix inside the start after the start time then exit
+            if self.turnpoints[t].kind == "start_exit": 
+                    if proceed_to_start:
+                        if not self.turnpoints[t].in_radius(fix):
+                            reached_turnpoints.append(fix)  #pilot has started
+                            t += 1
+                    if fix.rawtime > self.start_time and not proceed_to_start:
+                        if self.turnpoints[t].in_radius(fix):
+                            proceed_to_start = True         #pilot is inside start after the start time.
+                        
+            if self.turnpoints[t].kind == "start_enter":  #pilot must have at least 1 fix outside the start after the start time then enter
+                if proceed_to_start:
+                    if self.turnpoints[t].in_radius(fix):
+                        reached_turnpoints.append(fix)  #pilot has started
+                        t += 1
+                if fix.rawtime > self.start_time and not proceed_to_start:   
+                    if not self.turnpoints[t].in_radius(fix):
+                        proceed_to_start = True         #pilot is outside start after the start time.    
+            
+            if self.turnpoints[t].kind in ["cylinder", "End_of_speed_section", "goal_cylinder"]:
+                if self.turnpoints[t].in_radius(fix):
+                    reached_turnpoints.append(fix)  #pilot has achieved turnpoint
+                    t += 1
```

Comment:
wouldn't that then avoid evaluating the next turnpoint before the fix increments?
In reality unless the task setters make a mistake one fix should never achieve more than one turnpoint however. But in theory it could be possible.